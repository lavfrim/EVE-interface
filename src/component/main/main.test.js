import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

import Main from './index';

function Component() {
    return (
        <p>
            {'smth'}
        </p>
    );
}

const ArrayOfComponent = [
    <Component key={1}/>,
    <Component key={2} />,
    <Component key={3}/>
];

const mockStore = configureMockStore();
const store = mockStore({
    fractionsComponentArray: ArrayOfComponent,
});

describe('Main should render correctly', () => {
    it('with moch store', () => {
        const wrapper = mount(
            <Main store={store} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

