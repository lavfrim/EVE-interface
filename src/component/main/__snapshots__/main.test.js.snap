// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly with error mesage 1`] = `
<Connect(Main)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Main
    fractionsComponentArray={
      Array [
        <Component />,
        <Component />,
        <Component />,
      ]
    }
    setFractions={[Function]}
    setMessage={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <main
      className="main"
    >
      <Carousel
        amountCards={3}
        fractionsComponentArray={
          Array [
            <Component />,
            <Component />,
            <Component />,
          ]
        }
      >
        <section
          className="carousel"
        >
          <button
            className="carousel__button-backward"
            onClick={[Function]}
          >
            &lt;
          </button>
          <Component
            key="1"
          >
            <p>
              smth
            </p>
          </Component>
          <Component
            key="2"
          >
            <p>
              smth
            </p>
          </Component>
          <Component
            key="3"
          >
            <p>
              smth
            </p>
          </Component>
          <button
            className="carousel__button-forward"
            onClick={[Function]}
          >
            &gt;
          </button>
        </section>
      </Carousel>
    </main>
  </Main>
</Connect(Main)>
`;

exports[`Main should render correctly with mo 1`] = `
<Connect(Main)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Main
    fractionsComponentArray={
      Array [
        <Component />,
        <Component />,
        <Component />,
      ]
    }
    setFractions={[Function]}
    setMessage={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <main
      className="main"
    >
      <Carousel
        amountCards={3}
        fractionsComponentArray={
          Array [
            <Component />,
            <Component />,
            <Component />,
          ]
        }
      >
        <section
          className="carousel"
        >
          <button
            className="carousel__button-backward"
            onClick={[Function]}
          >
            &lt;
          </button>
          <Component
            key="1"
          >
            <p>
              smth
            </p>
          </Component>
          <Component
            key="2"
          >
            <p>
              smth
            </p>
          </Component>
          <Component
            key="3"
          >
            <p>
              smth
            </p>
          </Component>
          <button
            className="carousel__button-forward"
            onClick={[Function]}
          >
            &gt;
          </button>
        </section>
      </Carousel>
    </main>
  </Main>
</Connect(Main)>
`;

exports[`Main should render correctly with moch store 1`] = `
<Connect(Main)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Main
    fractionsComponentArray={
      Array [
        <Component />,
        <Component />,
        <Component />,
      ]
    }
    setFractions={[Function]}
    setMessage={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <main
      className="main"
    >
      <Carousel
        amountCards={3}
        fractionsComponentArray={
          Array [
            <Component />,
            <Component />,
            <Component />,
          ]
        }
      >
        <section
          className="carousel"
        >
          <button
            className="carousel__button-backward"
            onClick={[Function]}
          >
            &lt;
          </button>
          <Component
            key="1"
          >
            <p>
              smth
            </p>
          </Component>
          <Component
            key="2"
          >
            <p>
              smth
            </p>
          </Component>
          <Component
            key="3"
          >
            <p>
              smth
            </p>
          </Component>
          <button
            className="carousel__button-forward"
            onClick={[Function]}
          >
            &gt;
          </button>
        </section>
      </Carousel>
    </main>
  </Main>
</Connect(Main)>
`;
